common/src/KokkosFFT_padding.hpp

```diff
@@ -349,7 +235,11 @@ void crop_or_pad(const ExecutionSpace& exec_space, const InViewType& in,
       "type (float/double), the same layout (LayoutLeft/LayoutRight), and the "
       "same rank. ExecutionSpace must be accessible to the data in InViewType "
       "and OutViewType.");
-  crop_or_pad_impl(exec_space, in, out, s);
+  if constexpr (InViewType::rank() == 1) {
+    crop_or_pad_1D(exec_space, in, out);
+  } else {
+    crop_or_pad_ND(exec_space, in, out);
+  }
```

Comment:
```suggestion
  crop_or_pad(exec_space, in, out);
```

Can we implement such a function ? Either with a new function or merge the two existing if possible ?